synapse/api/constants.py

```diff
@@ -106,6 +106,9 @@ class EventTypes:
     MSC1772_SPACE_CHILD = "org.matrix.msc1772.space.child"
     MSC1772_SPACE_PARENT = "org.matrix.msc1772.space.parent"
 
+    MSC2716_INSERTION = "org.matrix.msc2716.insertion"
+    MSC2716_MARKER = "org.matrix.msc2716.marker"
```

Comment:
For Beeper who plans to use this after it merges: How do the prefixed "insertion"/"marker" events work after we unprefix?

It seems like we won't look for the prefixed events in the future and if we rely on them to navigate historical messages during the scrollback, we will have problems retrieving historical messages the scrollback.

I think we will be fine since "insertion"/"marker" events aren't needed to retrieve the messages (for the one local server) as we've placed them in the DAG (`depth`/`stream_order`) so they are navigable in the scrollback already without the "insertion"/"marker" events.

Reply:
Good question. It will be fine so long as they don't care about remote servers being able to pull in the history, which I think might be fine for their use case at least?